src/xenia/gpu/vulkan/vulkan_command_processor.cc

```diff
@@ -854,7 +856,7 @@ bool VulkanCommandProcessor::PopulateSamplers(VkCommandBuffer command_buffer,
 
   std::vector<xe::gpu::Shader::TextureBinding> dummy_bindings;
   auto descriptor_set = texture_cache_->PrepareTextureSet(
-      setup_buffer, current_batch_fence_, vertex_shader->texture_bindings(),
+      command_buffer, current_batch_fence_, vertex_shader->texture_bindings(),
```

Comment:
Q: Why did you change this to command_buffer? AFAIK uploads cannot be completed inside of a render pass.

Reply:
The setup_buffer is built through the entire frame and executed in its entirety before the command_buffer. The blits that we add here for carving up the resolve textures depend on the render passes completed in command_buffer; putting them in setup_buffer would have the blits execute before the render targets they read from are written.

Reply:
Okay, that's fair but you'll need to end the render pass when doing an upload/blit (as per vulkan spec).

I'd say pass both the setup buffer and command buffer, do uploads from memory in the setup buffer and uploads from other textures in the command buffer.